### ISSUE TITLE
error "file for role 'user' must contain a list of tasks" on simple role

### ISSUE THREAD

**Issue by agaffney** [ansible/ansible#13513]
I have a simple playbook with a few simple roles that work properly in 1.9.4. In both 2.0.0-0.7.rc2 (built via 'make deb') and a nightly build from this morning, I get this error:

```
$ ansible-playbook --check local.yml 
[DEPRECATION WARNING]: Instead of sudo/sudo_user, use become/become_user and make sure become_method is 'sudo' (default). This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
ERROR! The tasks/main.yml file for role 'user' must contain a list of tasks

The error appears to have been in '/tmp/ansible_stuff/roles/user/tasks/main.yml': line 3, column 1, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

# tasks file for user
- name: Say hello
^ here
```

The entire contents of this file are:

```

---
# tasks file for user
- name: Say hello
  debug: Hello, world!
```

This is with Python 2.7.9 on Debian Jessie-ish (it's really Raspbian).

**Comment by agaffney** [comment]
The same thing happens if I use `ansible-galaxy init foo` and then add the following to `roles/foo/tasks/main.yml` and `foo` to roles in the playbook:

```
- name: Foo
  debug: Foo
```

**Comment by agaffney** [comment]
The playbook is as follows:

```
$ cat local.yml 
---
- hosts: local
  sudo: true
  roles:
  - user
```

**Comment by agaffney** [comment]
I created a brand new setup to test this.

```
$ mkdir /tmp/ansible_test
$ cd /tmp/ansible_test
$ cat > hosts
[local]
localhost ansible_connection=local
^D
$ cat > local.yml
---
- hosts: local
  sudo: true
  roles:
  - foo
^D
$ mkdir roles
$ cd roles
$ ansible-galaxy init foo
$ echo -e "- name: Say hello\n  debug: Hello" >> foo/tasks/main.yml
$ cd ..
```

The problem still occurs:

```
$ ansible-playbook --check -i hosts local.yml
[DEPRECATION WARNING]: Instead of sudo/sudo_user, use become/become_user and make sure become_method is 'sudo' (default). This feature will be removed in a future release. 
Deprecation warnings can be disabled by setting deprecation_warnings=False in ansible.cfg.
ERROR! The tasks/main.yml file for role 'foo' must contain a list of tasks

The error appears to have been in '/tmp/ansible_test/roles/foo/tasks/main.yml': line 3, column 1, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

# tasks file for foo
- name: Say hello
^ here
```

**Comment by agaffney** [comment]
I commented out the try/except around the loading of the tasks list in `playbook/role/__init__.py`, and now I get this error:

```
ERROR! The requested action was not found in configured module paths. Additionally, core modules are missing. If this is a checkout, run 'git submodule update --init --recursive' to correct this problem.
```

**Comment by agaffney** [comment]
It appears that `/usr/share/pyshared/ansible/modules` is empty aside from a boilerplate `__init__.py` file. This is looking more like a problem with `make deb`.

**Comment by agaffney** [comment]
When I comment out the try/except on the nightly DEB, I get this error:

```
ERROR! this task 'debug' has extra params, which is only allowed in the following modules: command, shell, script, include, include_vars, add_host, group_by, set_fact, raw, meta
```

If I change my task to:

```
- name: Hello
  debug: msg=Hello
```

everything seems to be happy. I guess this is a case where 2.0 is most strict with 1.9.4.

**Comment by agaffney** [comment]
The 2 _actual_ issue here are:
- the github release tarballs apparently don't contain the core modules
- the try/except around loading tasks/handlers in `playbook/role/__init__.py` is hiding more useful error messages

**Comment by agaffney** [comment]
The code that's masking the real errors was apparently added very recently in https://github.com/ansible/ansible/commit/ce54a59cdc458add21cfee3750e21f21b25ad6fb

**Comment by bcoca** [comment]
hmm, seems like just adding AssertionError should fix the one issue

**Comment by bcoca** [comment]
can you test with this patch?

```
diff --git a/lib/ansible/playbook/role/__init__.py b/lib/ansible/playbook/role/__init__.py
index 1c6b344..f308954 100644
--- a/lib/ansible/playbook/role/__init__.py
+++ b/lib/ansible/playbook/role/__init__.py
@@ -150,7 +150,7 @@ class Role(Base, Become, Conditional, Taggable):
         current_when = getattr(self, 'when')[:]
         current_when.extend(role_include.when)
         setattr(self, 'when', current_when)
-        
+
         current_tags = getattr(self, 'tags')[:]
         current_tags.extend(role_include.tags)
         setattr(self, 'tags', current_tags)
@@ -174,7 +174,7 @@ class Role(Base, Become, Conditional, Taggable):
         if task_data:
             try:
                 self._task_blocks = load_list_of_blocks(task_data, play=self._play, role=self, loader=self._loader)
-            except:
+            except AssertionError:
                 raise AnsibleParserError("The tasks/main.yml file for role '%s' must contain a list of tasks" % self._role_name , obj=task_data)

         handler_data = self._load_role_yaml('handlers')
```

**Comment by agaffney** [comment]
That appears to work properly. I get the expected error message in my broken 2.0-rc7 DEB build (no core modules) and the nightly DEB (bad param for debug module)

**Comment by bcoca** [comment]
ok, im at bad param also

**Comment by bcoca** [comment]
fixed via d9e510b19273d6a495e6694b6930e49de80f9500
